src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, ChangeEvent, useEffect } from "react";
+import { getCurrentWeek } from "../../utils/dateUtils";
+import MealInputCell from "../MealInputCell/MealInputCell";
+import { mealTypes } from "../../types/types";
+import "./styles.scss";
+
+const Calendar: React.FC = () => {
+  const currentWeekDays = getCurrentWeek();
+
+  const [inputs, setInputs] = useState<{ [key: string]: string }>(() => {
+    const savedInputs = localStorage.getItem("calendarInputs");
+    if (savedInputs) {
+      return JSON.parse(savedInputs);
+    }
+    return Object.fromEntries(
```

Comment:
this isn't 100% clear to me, u sure this is the simplest most readable way of getting your mapped array?